pkg/traceql/ast.go

```diff
@@ -197,6 +193,7 @@ func newGroupOperation(e FieldExpression) GroupOperation {
 
 func (o GroupOperation) extractConditions(request *FetchSpansRequest) {
 	o.Expression.extractConditions(request)
+	request.AllConditions = false
```

Comment:
```
{ foo } | by(span.bar)
```
can't have AllConditions = true? is this what we were discussing where we're not sure if the nil group is valid or not?

Reply:
Yes this is what we were discussing offline.  For this query `by(span.bar)` doesn't require span.bar to exist.   It's different than `{ ... && span.bar != nil }...`.   It is more like ` | select(span.bar)` except it can't actually go in the second pass.  Therefore it is achieved with AllConditions=false, and OpNone in the first pass.